test/gnmi/offlineintopotest.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package gnmi
+
+import (
+	"context"
+	"github.com/onosproject/onos-config/pkg/northbound/gnmi"
+	"github.com/onosproject/onos-test/pkg/onit/env"
+	"github.com/onosproject/onos-topo/api/device"
+	"github.com/openconfig/gnmi/proto/gnmi_ext"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	offlineInTopoModPath          = "/system/clock/config/timezone-name"
+	offlineInTopoModValue         = "Europe/Rome"
+	offlineInTopoModDeviceName    = "offline-dev-1"
+	offlineInTopoModDeviceVersion = "1.0.0"
+	offlineInTopoModDeviceType    = "Devicesim"
+)
+
+// TestOfflineDeviceInTopo tests set/query of a single GNMI path to a single device that is in the config but offline
+func (s *TestSuite) TestOfflineDeviceInTopo(t *testing.T) {
+
+	deviceClient, deviceClientError := env.Topo().NewDeviceServiceClient()
+	assert.NotNil(t, deviceClient)
+	assert.Nil(t, deviceClientError)
+	newDevice := &device.Device{
+		ID:          offlineInTopoModDeviceName,
+		Revision:    0,
+		Address:     offlineInTopoModDeviceName + ":10161",
+		Target:      "",
+		Version:     offlineInTopoModDeviceVersion,
+		Timeout:     nil,
+		Credentials: device.Credentials{},
+		TLS:         device.TlsConfig{},
+		Type:        offlineInTopoModDeviceType,
+		Role:        "",
+		Attributes:  nil,
+		Protocols:   nil,
+	}
+	addRequest := &device.AddRequest{Device: newDevice}
+	addResponse, addResponseError := deviceClient.Add(context.Background(), addRequest)
+	assert.NotNil(t, addResponse)
+	assert.Nil(t, addResponseError)
+
+	simulator := env.NewSimulator().SetName(offlineInTopoModDeviceName)
+
+	// Make a GNMI client to use for requests
+	c, err := env.Config().NewGNMIClient()
+	assert.NoError(t, err)
+	assert.True(t, c != nil, "Fetching client returned nil")
+
+	// Set a value using gNMI client to the offline device
+	extNameDeviceType := gnmi_ext.Extension_RegisteredExt{
```

Comment:
I don't think you need to give these `DeviceType` and `DeviceVersion` extensions here  - because the Device is in Topo already and the type and version are already known. 

You could add the Network Change Id (100) though here and then you would have a predictable name for the Network change below.